base/compiler/ssair/legacy.jl

```diff
@@ -57,6 +57,12 @@ function replace_code_newstyle!(ci::CodeInfo, ir::IRCode, nargs::Int)
     ci.linetable = ir.linetable
     ci.ssavaluetypes = ir.types
     ci.ssaflags = ir.flags
+    for metanode in ir.meta
```

Comment:
Would be good to have this separate and with a test that the `noinline` function attribute is now set again.

Reply:
You mean a separate PR just with the meta node fix?

Reply:
Yes :) that should be very easy to merge quickly while we figure out the bigger picture here

Reply:
Alright, I'll prepare it. I may need some more help writing the test, though.

Reply:
I guess this fixes https://github.com/JuliaLang/julia/issues/27104?

Reply:
https://github.com/JuliaLang/julia/pull/31871

Reply:
Merged.